packages/orchestrator/lib/clients/processor.ts

```diff
@@ -141,8 +143,18 @@ export class OrchestratorProcessor {
                 }
             } finally {
                 this.abortControllers.delete(task.id);
+                clearInterval(heartbeat);
                 span.finish();
             }
         });
     }
+
+    private heartbeat(task: OrchestratorTask): NodeJS.Timeout {
+        return setInterval(async () => {
+            const res = await this.orchestratorClient.heartbeat({ taskId: task.id });
+            if (res.isErr()) {
+                logger.error(`failed to send heartbeat for task ${task.id}: ${stringifyError(res.error)}`);
+            }
+        }, 300_000);
```

Comment:
that's a long delay between heartbeat no? I would say 60s is already "huge"

Reply:
This is the current setting. I think we should maintain the current setting to keep the number of moving parts to a minimum and change this if we think it is necessary later.

If there is suddenly an issue with not using temporal anymore this is one less thing we have to check on that was changed that could be causing issues.

Reply:
it is a long delay between heartbeat indeed but like Khaliq said I didn't want to change too many things. I am happy to lower it once we temporal is removed